tests/js/client/shell/api/api-replication.js

```diff
@@ -0,0 +1,1373 @@
+      }
+    },
+
+    testRestoreEdge: function () {
+      const res = tryRestore({name: collname, type: 3});
+      try {
+        assertTrue(res.result, `Result: ${JSON.stringify(res)}`);
+        validateProperties({}, collname, 3);
+      } finally {
+        db._drop(collname);
+      }
+    },
+
+    testRestoreDocument: function () {
+      const res = tryRestore({name: collname, type: 2});
+      try {
+        assertTrue(res.result, `Result: ${JSON.stringify(res)}`);
+        validateProperties({}, collname, 2);
+      } finally {
+        db._drop(collname);
+      }
+    },
+
+    testRestoreIllegalType: function () {
+      const illegalType = [
+        0,
+        1,
+        4
+      ];
+
+      for (const type of illegalType) {
+        const res = tryRestore({name: collname, type});
+        assertEqual(res.code, ERROR_HTTP_BAD_PARAMETER.code, `Could create ${type}`);
+        assertEqual(res.errorNum, ERROR_ARANGO_COLLECTION_TYPE_INVALID.code);
+      }
+
+      const ignoredType = [
+        "test",
+        "document",
+        "edge",
+        {},
+        [],
+        true
+      ];
+
+      // ill-formatted types are ignored, and fall back to document
+      for (const type of ignoredType) {
+        const res = tryRestore({name: collname, type});
+        try {
+          assertTrue(res.result, `Result: ${JSON.stringify(res)}`);
+          validateProperties({}, collname, 2);
+        } finally {
+          db._drop(collname);
+        }
+      }
+    },
+
+    testRestoreDeleted: function () {
+      const res = tryRestore({name: collname, type: 2, deleted: true});
+      try {
+        assertTrue(res.result, `Result: ${JSON.stringify(res)}`);
+        assertEqual(db._collections().filter((c) => c.name() === collname).length, 0, `Deleted collection was restored.`);
+      } finally {
+        db._drop(collname);
+      }
+    },
+
+    testRestoreNumberOfShards: function () {
+      const res = tryRestore({name: collname, type: 2, numberOfShards: 3});
+      try {
+        assertTrue(res.result, `Result: ${JSON.stringify(res)}`);
+        validateProperties({numberOfShards: 3}, collname, 2);
+      } finally {
+        db._drop(collname);
+      }
+    },
+
+    testRestoreListOfShards: function () {
+      const res = tryRestore({name: collname, shards: {"s01": ["server1", "server2"], "s02": ["server3"]}});
+      try {
+        assertTrue(res.result, `Result: ${JSON.stringify(res)}`);
+        validateProperties({numberOfShards: 2}, collname, 2);
+      } finally {
+        db._drop(collname);
+      }
+    },
+
+    testRestoreReplicationFactor: function () {
+      const res = tryRestore({name: collname, replicationFactor: 3});
+      try {
+        assertTrue(res.result, `Result: ${JSON.stringify(res)}`);
+        validateProperties({replicationFactor: 3}, collname, 2);
+      } finally {
+        db._drop(collname);
+      }
+    },
+
+    testRestoreReplicationFactorSatellite: function () {
```

Comment:
We should do something about this test, otherwise everybody is going to have a red CI. The following properties are extra on enterprise single-server: ` ["isSmart","shardKeys","numberOfShards","replicationFactor","isDisjoint"]`
We could:
1. Add a separate case for single server and add these properties to the override parameter, such that we expect them.
2. Find out why there are more properties in this case and fix that, if it is even possible.
3. Disable the test.